emulator/src/kreator/assembler.rs

```diff
@@ -69,10 +69,12 @@ impl Assembler {
         Ok(byte_code)
     }
 
-    pub fn get_line_map(&self) -> Result<Vec<(u16, usize)>, &'static str> {
+    pub fn get_line_map(&self) -> Result<Vec<usize>, &'static str> {
         let mapping = get_line_map(&self.code)?;
-        let mut mapped_vec = mapping.into_iter().collect::<Vec<(u16, usize)>>();
-        mapped_vec.sort_by(|(a,_), (b,_)| a.cmp(b));
+        let mut mapped_vec: Vec<usize> = vec![0; mapping.len()];
+        for (byte, line) in mapping {
+            *mapped_vec.get_mut(usize::from(byte)).unwrap() = line;
```

Comment:
I would recommend using standard indexing here, aka:

`mapped_vec[byte as usize] = line;

Reply:
Seems like a good call, I don't know why I didn't do it like that in the first place.